docs/secret.md

```diff
@@ -0,0 +1,171 @@
+# Secret
+
+When you are defining a datasource, likely you will need to provide a basic authentication, a certificate, or a token
+to be used when the Perses backend will contact your Datasource.
+
+`Secret` or `GlobalSecret` are the two objects that you need. The choice between a `Secret` or
+a `GlobalSecret` will mainly depend on if you are defining a `Datasource` or a `GlobalDatasource`.
```

Comment:
Maybe we should highlight more the difference in terms of "scope behavior" with the rest of the resources here?
I mean e.g a project variable can rely on a global datasource for instance, while here we have:
- GlobalDatasource -> GlobalSecret
- (Project) Datasource -> Secret only
- Dashboard datasource -> didnt check but Secret only too I guess ?

Maybe it's an issue to have such discrepancy btw, not sure but anyway that's out of the topic of this PR.

WDYT?

Reply:
ah yeah indeed it's certainly unclear like that. I will change that